### PR TITLE
Rover: Activation of OSD

### PR DESCRIPTION
**Pull request by Landflieger**
Rover: Activation of OSD for boards which are able to display it

hopefully this PR is way better ...

### CODE REVIEW

**Comment by rmackay9** [comment]
This looks pretty harmless, I guess it works?  I see it's missing the publish_osd_info() function that Plane and Copter have so perhaps some of the waypoint info will be missing.  Still, I guess that could come later.

I'm happy for the OSD to be enabled by default if you like.

**Comment by Landflieger** [comment]
yes, of course it works! ;) Tested on MatekF405-Wing and SITL (more I couldn't do).
Yes publish_osd_info() I left out intentionally... I#m not sure if drivers really need that info and I don't want an information overkill... we can add it later ...

yes, I think it is a good idea to enable it by default... thank you !

**Comment by WickedShell** [comment]
@rmackay9 `OSD_ENABLED` is set by the HAL if the chip is there, you want to disable it by default to save space & parameters on a board that doesn't have the required chips.

File: APMrover2/Parameters.h [hunk 1]
```diff
@@ -38,6 +38,8 @@ class Parameters {
         k_param_pivot_turn_angle,
         k_param_rc_13_old,  // unused
         k_param_rc_14_old,  // unused
+        k_param_osd,
```
Comment (tridge) [inline APMrover2/Parameters.h]: this would set k_param_osd to 20, which is already used

File: APMrover2/Parameters.cpp [hunk 2]
```diff
@@ -413,6 +413,12 @@ const AP_Param::Info Rover::var_info[] = {
     // @Path: Parameters.cpp
     GOBJECT(g2, "",  ParametersG2),
 
+    #if OSD_ENABLED
```
Comment (tridge) [inline APMrover2/Parameters.cpp]: this should be "#if OSD_ENABLED == ENABLED"

File: APMrover2/config.h [hunk 3]
```diff
@@ -170,4 +170,7 @@
 #else
  #define DEVO_TELEM_ENABLED ENABLED
 #endif
+#ifndef OSD_ENABLED
```
Comment (amilcarlucas) [inline APMrover2/config.h]: OSD_ENABLED == ENABLED

Comment (sh83) [reply APMrover2/config.h]: I think here should be ifndef, not comparation.

File: APMrover2/Parameters.h (hunk 1 above)
Comment (Landflieger) [reply APMrover2/Parameters.h]: I will try to fix it later, just back from office ...

File: APMrover2/config.h (hunk 3 above)
Comment (Landflieger) [reply APMrover2/config.h]: ifndef is also used in ArduPlane/config.h

File: APMrover2/Parameters.cpp (hunk 2 above)
Comment (Landflieger) [reply APMrover2/Parameters.cpp]: I wiöö change it ...

**Comment by Landflieger** [comment]
Didn't know why updated pr throws error on travis-cl test ... the affected errors come from modules which aren't affected by pr ...

could somebody please give me some hints how to correct it ... thx

**Comment by Landflieger** [comment]
@tridge fixed missing new line

**Comment by tridge** [comment]
I've forced pushed some fixups for the issues I found to the PR branch

**Comment by Landflieger** [comment]
thx

**Comment by tridge** [comment]
@Landflieger I was just waiting for CI to pass, merged now.
Thanks for the PR!

**Comment by Landflieger** [comment]
@tridge great! Thanks a lot ! Proud to contribute my first PR :)

**Pull request merged**